src/ucs/sys/iovec.inl

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCS_IOVEC_INL_
+#define UCS_IOVEC_INL_
+
+#include <ucs/sys/math.h>
+#include <ucs/sys/iovec.h>
+#include <ucs/debug/assert.h>
+
+
+/**
+ * Fill the destination array of IOVs by data provided in the source
+ * array of IOVs.
+ * The function avoids copying IOVs with zero length.
+ *
+ * @param [out]    _dst_iov              Pointer to the resulted array of IOVs.
+ * @param [in/out] _dst_iov_cnt_p        Pointer to the varibale that holds the number
+ *                                       of the elements in the array of IOVs (input:
+ *                                       initial, out: result).
+ * @param [in]     _dst_iov_set_buffer_f Function that sets the buffer to the IOV element
```

Comment:
maybe pass field names instead of function pointers? this way you will not need to define setters and getters

Reply:
yes, it looks better, but doesn't work for UCT IOV that requires calling `uct_iov_get_length()`